warehouse/macaroons/caveats/__init__.py

```diff
@@ -149,6 +145,104 @@ def verify(self, request: Request, context: Any, permission: str) -> Result:
         return Success()
 
 
+class PublicPermissions(enum.IntFlag, boundary=enum.FlagBoundary.STRICT):
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
**praise:** loving the inline documentation here - it's a pretty confusing topic, so the more clarity, the better.‏